modules/shared/src/main/scala/org/tessellation/schema/SnapshotOrdinal.scala

```diff
@@ -4,16 +4,18 @@ import cats.Order
 import cats.kernel._
 import cats.syntax.semigroup._
 
-import derevo.cats.{order, show}
+import derevo.cats.{eqv, order, show}
 import derevo.derive
 import eu.timepit.refined.cats._
 import eu.timepit.refined.numeric.NonNegative
 import eu.timepit.refined.refineV
 import eu.timepit.refined.types.numeric.NonNegLong
 import io.circe.{Decoder, Encoder}
 
-@derive(order, show)
-case class SnapshotOrdinal(value: NonNegLong)
+@derive(order, show, eqv)
+case class SnapshotOrdinal(value: NonNegLong) {
+  def plus(addend: NonNegLong) = SnapshotOrdinal(value |+| addend)
+}
```

Comment:
you don't need `plus` method in here. `SnapshotOrdinal` already supports `Next` which allows you to increment also by `n` ordinals.